mongodb-datastore/handlers/events.go

```diff
@@ -114,64 +114,102 @@ func GetEvents(params event.GetEventsParams) (*event.GetEventsOKBody, error) {
 		searchOptions["data.service"] = params.Service
 	}
 
-	var newNextPageKey int64
-	var nextPageKey int64 = 0
-	if params.NextPageKey != nil {
-		tmpNextPageKey, _ := strconv.Atoi(*params.NextPageKey)
-		nextPageKey = int64(tmpNextPageKey)
-		newNextPageKey = nextPageKey + *params.PageSize
-	} else {
-		newNextPageKey = *params.PageSize
-	}
+	var result event.GetEventsOKBody
 
-	pageSize := *params.PageSize
-	sortOptions := options.Find().SetSort(bson.D{{"time", -1}}).SetSkip(nextPageKey).SetLimit(pageSize)
+	if params.Root != nil {
+		var values []interface{}
+		values, err = collection.Distinct(ctx, "shkeptncontext", searchOptions)
 
-	totalCount, err := collection.CountDocuments(ctx, searchOptions)
-	if err != nil {
-		logger.Error(fmt.Sprintf("error counting elements in events collection: %v", err))
-	}
+		if err != nil {
+			logger.Error(fmt.Sprintf("error loading distinct shkeptncontext: %v", err))
+		}
 
-	cur, err := collection.Find(ctx, searchOptions, sortOptions)
-	if err != nil {
-		logger.Error(fmt.Sprintf("error finding elements in events collection: %v", err))
-	}
+		for _, value := range values {
+			var outputEvent interface{}
+
+			sortOptions := options.FindOne().SetSort(bson.D{{"time", -1}})
```

Comment:
This might be wrong. I think setting the sort to `time` and `-1` means "decreasing", so you get the latest event. We need the earliest event.